src/Files.App/ViewModels/Properties/SecurityAdvancedViewModel.cs

```diff
@@ -0,0 +1,218 @@
+				else
+					return "SecurityAdvancedInheritedRemove/Text".GetLocalizedResource();
+			}
+		}
+
+		private bool _isProtected;
+
+		private bool _preserveInheritance;
+
+		private GridLength _columnType = new(64d);
+		public GridLength ColumnType
+		{
+			get => _columnType;
+			set => SetProperty(ref _columnType, value);
+		}
+
+		private GridLength _columnPrincipal = new(200d);
+		public GridLength ColumnPrincipal
+		{
+			get => _columnPrincipal;
+			set => SetProperty(ref _columnPrincipal, value);
+		}
+
+		private GridLength _columnAccess = new(160d);
+		public GridLength ColumnAccess
+		{
+			get => _columnAccess;
+			set => SetProperty(ref _columnAccess, value);
+		}
+
+		private GridLength _columnInherited = new(70d);
+		public GridLength ColumnInherited
+		{
+			get => _columnInherited;
+			set => SetProperty(ref _columnInherited, value);
+		}
+
+		public RelayCommand ChangeOwnerCommand { get; set; }
+		public RelayCommand AddAccessControlEntryCommand { get; set; }
+		public RelayCommand RemoveAccessControlEntryCommand { get; set; }
+		public RelayCommand DisableInheritanceCommand { get; set; }
+		public RelayCommand<string> SetDisableInheritanceOptionCommand { get; set; }
+		public RelayCommand ReplaceChildPermissionsCommand { get; set; }
+
+		public SecurityAdvancedViewModel(ListedItem item, Window window)
+		{
+			IsFolder = item.PrimaryItemAttribute == StorageItemTypes.Folder && !item.IsShortcut;
+			Item = item;
+			Window = window;
+
+			InitializeCommands();
+			GetAccessControlList();
+		}
+
+		public SecurityAdvancedViewModel(DriveItem item, Window window)
+		{
+			IsFolder = true;
+			Item = new ListedItem()
+			{
+				ItemNameRaw = item.Text,
+				ItemPath = item.Path,
+				PrimaryItemAttribute = StorageItemTypes.Folder
+			};
+			Window = window;
+
+			InitializeCommands();
+			GetAccessControlList();
+		}
+
+		private void InitializeCommands()
+		{
+			ChangeOwnerCommand = new RelayCommand(ChangeOwner, () => AccessControlList is not null);
+			AddAccessControlEntryCommand = new RelayCommand(AddAccessControlEntry, () => AccessControlList is not null && AccessControlList.IsValid);
+			RemoveAccessControlEntryCommand = new RelayCommand(RemoveAccessControlEntry, () => AccessControlList is not null && AccessControlList.IsValid && SelectedAccessControlEntry is not null);
+			DisableInheritanceCommand = new RelayCommand(DisableInheritance, () => AccessControlList is not null && AccessControlList.IsValid && (AccessControlList.IsProtected != _isProtected));
+			SetDisableInheritanceOptionCommand = new RelayCommand<string>(SetDisableInheritanceOption);
+			ReplaceChildPermissionsCommand = new RelayCommand(ReplaceChildPermissions, () => AccessControlList is not null && AccessControlList.IsValid);
+		}
+
+		private async void ChangeOwner()
+		{
+			var pickedObject = await OpenObjectPicker();
+			if (pickedObject is not null)
+			{
+				bool isFolder = Item.PrimaryItemAttribute == StorageItemTypes.Folder && !Item.IsShortcut;
+
```

Comment:
It might be better to use `string.IsNullOrEmpty` instead of just `is not null` (you would also be more consistent considered `AddAccessControlEntry()`)
What is `isFolder` needed for?

Reply:
As above